src/views/InventoryView.tsx

```diff
@@ -132,8 +132,21 @@ export class InventoryView extends React.Component<{ repo: IIngredientRepo & IRe
     }
 
     private renderInventoryRow(ingredient: Ingredient) {
+        let ingredientExpMarker = "";
+        if(ingredient.expires){
+            let d = new Date();
+            let conversion = 24*60*60*1000;
+            let shelf_life = ingredient.shelf_life || 0;
+            if(ingredient.last_purchased+conversion*(shelf_life-2) < d.getTime()){
```

Comment:
Can you comment out this date math a little bit, and move it all to a function like ingredient.IsExpiringSoon() or the like?

Reply:
Done